.github/workflows/release-python-client.yml

```diff
@@ -0,0 +1,57 @@
+name: Build and publish Python client to PyPI
+
+on:
+  pull_request:
+  workflow_call:
+    inputs:
+      new-tag:
+        required: true
+        type: string
+jobs:
+  build-n-publish:
+    name: Build and publish Python client to PyPI
+    runs-on: ubuntu-latest
+    permissions:
+      # IMPORTANT: this permission is mandatory for trusted publishing (OIDC)
+      # https://docs.pypi.org/trusted-publishers/using-a-publisher/
+      id-token: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.12
+
+      - name: Install openapi-generator-cli
+        run: npm install @openapitools/openapi-generator-cli -g
+
+      - name: set version to 7.8.0
+        run: openapi-generator-cli version-manager set 7.8.0
+
+      - name: Generate Python client
+        run: |
+          openapi-generator-cli generate \
+          -i common/openapi.yml \
+          -g python \
+          -o teaspoons-client \
+          --additional-properties=projectName=terra-scientific-pipelines-service-client,packageName=teaspoons_client,packageVersion=${{ inputs.new-tag || '0.0.0' }},httpUserAgent=terra-scientific-pipeliens-service-client/${{ inputs.new-tag || '0.0.0'}}/python
```

Comment:
Do we call this `terra-scientific-pipelines-service-thin-client` or how will we distinguish this Pypi package from the thick client?